packages/database/src/prisma-api.ts

```diff
@@ -456,7 +466,7 @@ export function getSdk(
         quantity > 1 ||
         Boolean(existingBulkCoupon) ||
         options.bulk ||
-        Boolean(existingPurchase)
+        Boolean(existingPurchase?.status === 'Valid')
```

Comment:
only do this for a valid purchase otherwise it treats rfunded/restricted/etc as "bulk" which isn't what we want

Reply:
It was already restricting the lookup of the `existingPurchase` by `status`, so this is covered. This is the only place that `existingPurchase` is checked, so the original query could continue to look up only `Valid` purchases and this addendum (`existingPurchase?.status === 'Valid'`) wouldn't be needed.

<img width="542" alt="CleanShot 2023-03-06 at 09 46 47@2x" src="https://user-images.githubusercontent.com/694063/223160102-57ef4fe4-033c-477c-9919-6b8089f53d32.png">